### ISSUE TITLE
Permissions error on Marshmallow

### ISSUE THREAD

**Issue by steffandroid** [auth0/Lock-GooglePlus.Android#12]
When targeting API level 23 and attempting to use Google+ for signing in, I'm getting this error message:

![Auth0 Error](http://i.imgur.com/qir14C4.png)

With this stacktrace:

```
E/com.auth0.lock.LockActivity: Failed to authenticate user
E/com.auth0.lock.LockActivity: java.lang.SecurityException: Missing android.permission.GET_ACCOUNTS
E/com.auth0.lock.LockActivity:     at android.os.Parcel.readException(Parcel.java:1599)
E/com.auth0.lock.LockActivity:     at android.os.Parcel.readException(Parcel.java:1552)
E/com.auth0.lock.LockActivity:     at com.google.android.gms.plus.internal.zzd$zza$zza.getAccountName(Unknown Source)
E/com.auth0.lock.LockActivity:     at com.google.android.gms.plus.internal.zze.getAccountName(Unknown Source)
E/com.auth0.lock.LockActivity:     at com.google.android.gms.internal.zzpn.getAccountName(Unknown Source)
E/com.auth0.lock.LockActivity:     at com.auth0.googleplus.FetchTokenAsyncTask.doInBackground(FetchTokenAsyncTask.java:63)
E/com.auth0.lock.LockActivity:     at com.auth0.googleplus.FetchTokenAsyncTask.doInBackground(FetchTokenAsyncTask.java:46)
E/com.auth0.lock.LockActivity:     at android.os.AsyncTask$2.call(AsyncTask.java:295)
E/com.auth0.lock.LockActivity:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/com.auth0.lock.LockActivity:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
E/com.auth0.lock.LockActivity:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
E/com.auth0.lock.LockActivity:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
E/com.auth0.lock.LockActivity:     at java.lang.Thread.run(Thread.java:818)
```

This is because the GET_ACCOUNTS permission is required, and this is granted at runtime in Android Marshmallow. I could request this permission before starting LockActivity, but I'd argue it's far better UX for the user to be prompted for this permission only if they click on the G+ icon. Could you add this functionality?

**Comment by nikolaseu** [comment]
It's in our TODO, but low priority. Meanwhile you could just compile targeting API 22
